### PR TITLE
fix: use zksync contract size limits in --zksync mode

### PR DESCRIPTION
**Pull request by nbaztec**
## Motivation
As reported in #504 we are still reporting the size margins from [eip-170](https://eips.ethereum.org/EIPS/eip-170) (foundry default) in zksync mode.

## Solution
Report margins via zksync's contract sizes as stated in their [documentation](https://docs.zksync.io/build/developer-reference/ethereum-differences/contract-deployment#contract-size-limit-and-format-of-bytecode-hash).

### CODE REVIEW

**Review by elfedy — APPROVED** [review]

File: .github/workflows/test.yml
Comment (Karrq) [inline .github/workflows/test.yml:1]: Are we sure we want to disable zk-tests with this PR? considering that we are still migrating to cargo test

File: crates/forge/tests/cli/build.rs [hunk 1]
```diff
@@ -51,6 +51,21 @@ forgetest_init!(build_sizes_no_forge_std, |prj, cmd| {
...
 
+// tests build output is as expected in zksync mode
+forgetest_init!(test_zk_dbg_build_sizes, |prj, cmd| {
+    use std::process::Command;
+    let output = Command::new("sh")
+        .arg("-c")
+        .arg("ls -l /home/runner/.zksync/zksolc-linux-amd64-musl-v1.5.1")
+        .output()
+        .expect("failed to execute process");
+    println!("{output:?}");
```
Comment (Karrq) [inline crates/forge/tests/cli/build.rs:62]: Do we want to do this check here or just let foundry do it? How's this portable with non-ci environments?

**Review by Karrq — COMMENTED** [review]
Why are we renaming some of the zk tests to have `dbg` in their name?

File: .github/workflows/test.yml
Comment (nbaztec) [reply .github/workflows/test.yml:1]: cause I'm debugging a parallel test issue atm

File: crates/forge/tests/cli/build.rs (hunk 1 above)
Comment (nbaztec) [reply crates/forge/tests/cli/build.rs:62]: see above

File: .github/workflows/test.yml
Comment (Karrq) [reply .github/workflows/test.yml:1]: so we shouldn't merge this PR yet?

Comment (nbaztec) [reply .github/workflows/test.yml:1]: No not yet

**Pull request merged**